java/daikon/dcomp/DCInstrument.java

```diff
@@ -1935,8 +2076,12 @@ InstructionList xform_inst_refs_only(MethodGen mg, InstructionHandle ih, Operand
    * Adds a call to DCruntime.exit() at each return from the method. This call calculates
    * comparability on the daikon variables. It is only necessary if we are tracking comparability
    * for the variables of this method.
+   *
+   * @param mg method to modify
+   * @param mi MethodInfo for method
+   * @param method_info_index index for MethodInfo
    */
-  public void add_exit(MethodGen mg, MethodInfo mi, int method_info_index) {
+  void add_exit(MethodGen mg, MethodInfo mi, int method_info_index) {
```

Comment:
Many methods in DCInstrument were changed from `public` to package-protected.  (That was the reason for the need for a lot of Javadoc.)  I'm curious about the reason for this change.

Reply:
The methods modified are all defined and used entirely within DCInstrument.   I couldn't see a use case for them to be public.

Reply:
I see.  In that case, should they be private?

Reply:
The coding style seemed to be use private for data but not methods.  There are a few exceptions, of course.  I'm happy to change to a standard/recommended style.

Reply:
A general rule of thumb is to make all data private, to avoid data corruption by clients directly changing fields without maintaining invariants.

Methods should be made public/protected/private/package-private depending on their intended use; there isn't a general rule for Java (nor one for this codebase that I am aware of) that requires methods not to be private.